src/overlays/actors/ovl_En_Bal/z_en_bal.c

```diff
@@ -15,7 +33,91 @@ void EnBal_Destroy(Actor* thisx, PlayState* play);
 void EnBal_Update(Actor* thisx, PlayState* play);
 void EnBal_Draw(Actor* thisx, PlayState* play);
 
-#if 0
+typedef enum EnBalEyeTexture {
+    /* 0x0 */ EN_BAL_EYETEX_OPEN,
+    /* 0x1 */ EN_BAL_EYETEX_CLOSED
+} EnBalEyeTexture;
+
+typedef enum EnBalAnimation {
+    /* 0x0 */ EN_BAL_ANIM_FLOAT_DRAW,
+    /* 0x1 */ EN_BAL_ANIM_FALL_LOOP,
+    /* 0x2 */ EN_BAL_ANIM_FALL_ONCE,
+    /* 0x3 */ EN_BAL_ANIM_LAND,
+    /* 0x4 */ EN_BAL_ANIM_TALK_TWIST,
+    /* 0x5 */ EN_BAL_ANIM_TALK_IDLE,
+    /* 0x6 */ EN_BAL_ANIM_MAGIC_FULL,
+    /* 0x7 */ EN_BAL_ANIM_HAPPY_DANCE_LOOP,
+    /* 0x8 */ EN_BAL_ANIM_HAPPY_DANCE_ONCE,
+    /* 0x9 */ EN_BAL_ANIM_MAGIC_SHORT,
+    /* 0xA */ EN_BAL_ANIM_WAIT,
+    /* 0xB */ EN_BAL_ANIM_SPIN,
+    /* 0xC */ EN_BAL_ANIM_HIDE_FACE,
+    /* 0xD */ EN_BAL_ANIM_CONFETTI
+} EnBalAnimation;
+
+typedef enum EnBalBuyMapChoice {
+    /* 0x0 */ EN_BAL_MAPCHOICE_PROXIMAL,
+    /* 0x1 */ EN_BAL_MAPCHOICE_DISTAL,
+    /* 0x2 */ EN_BAL_MAPCHOICE_CANCEL
+} EnBalBuyMapChoice;
+
+typedef enum EnBalWatchTarget {
+    /* 0x0 */ EN_BAL_WATCHTRG_NONE,
+    /* 0x1 */ EN_BAL_WATCHTRG_PLAYER,
+    /* 0x2 */ EN_BAL_WATCHTRG_FAIRY
+} EnBalWatchTarget;
+
+typedef enum EnBalBalloonAction {
+    /* 0x0 */ EN_BAL_BALACT_NONE,
+    /* 0x1 */ EN_BAL_BALACT_POP,
+    /* 0x2 */ EN_BAL_BALACT_FALL,
+    /* 0x4 */ EN_BAL_BALACT_INFLATE = 4,
+    /* 0x5 */ EN_BAL_BALACT_RISE
+} EnBalBalloonAction;
+
+typedef enum EnBalIdleAnimStage {
+    /* 0x0 */ EN_BAL_IDLESTAGE_ACTIVITY,
+    /* 0x2 */ EN_BAL_IDLESTAGE_PREP_WAIT = 2,
+    /* 0x3 */ EN_BAL_IDLESTAGE_WAIT
+} EnBalIdleAnimStage;
+
+void EnBal_SetMainColliderToBalloon(EnBal* this);
+void EnBal_SetMainColliderToHead(EnBal* this);
+s32 EnBal_ValidatePictograph(PlayState* play, Actor* thisx);
```

Comment:
Many of these functions don't need forward declaration. If the function is written above where it's called, could you remove the forward declarations?